processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -137,14 +137,12 @@ public BeanMappingMethod build() {
                 return null;
             }
 
-            if ( !method.getMappingOptions().isRestrictToDefinedMappings() ) {
```

Comment:
I think you're right for this case 👍 

But now consider:

`@Mapping(target = "song.artist.label.studio.city", source = "chartEntry1.city" ),`

Suppose there would be a property: `number` in `studio` and a `number` in `chartEntry1` I would not expect them to be mapped automatically, right? And which one would be valid? the one from `chartEntry1` or `chartEntry2`?